wrappers/javascript/anoncreds-shared/src/api/Credential.ts

```diff
@@ -1,66 +1,138 @@
-import type { CredentialDefinition } from './CredentialDefinition'
-import type { CredentialDefinitionPrivate } from './CredentialDefinitionPrivate'
-import type { CredentialOffer } from './CredentialOffer'
-import type { CredentialRequest } from './CredentialRequest'
-import type { CredentialRequestMetadata } from './CredentialRequestMetadata'
+import type { ObjectHandle } from '../ObjectHandle'
+import type { JsonObject } from '../types'
 import type { CredentialRevocationConfig } from './CredentialRevocationConfig'
-import type { MasterSecret } from './MasterSecret'
-import type { RevocationRegistryDefinition } from './RevocationRegistryDefinition'
-import type { RevocationStatusList } from './RevocationStatusList'
 
 import { AnoncredsObject } from '../AnoncredsObject'
 import { anoncreds } from '../register'
 
+import { CredentialDefinition } from './CredentialDefinition'
+import { CredentialDefinitionPrivate } from './CredentialDefinitionPrivate'
+import { CredentialOffer } from './CredentialOffer'
+import { CredentialRequest } from './CredentialRequest'
+import { CredentialRequestMetadata } from './CredentialRequestMetadata'
+import { MasterSecret } from './MasterSecret'
+import { RevocationRegistryDefinition } from './RevocationRegistryDefinition'
+import { RevocationStatusList } from './RevocationStatusList'
+import { pushToArray } from './utils'
+
 export type CreateCredentialOptions = {
-  credentialDefinition: CredentialDefinition
-  credentialDefinitionPrivate: CredentialDefinitionPrivate
-  credentialOffer: CredentialOffer
-  credentialRequest: CredentialRequest
+  credentialDefinition: CredentialDefinition | JsonObject
+  credentialDefinitionPrivate: CredentialDefinitionPrivate | JsonObject
+  credentialOffer: CredentialOffer | JsonObject
+  credentialRequest: CredentialRequest | JsonObject
   attributeRawValues: Record<string, string>
   attributeEncodedValues?: Record<string, string>
   revocationRegistryId?: string
   revocationConfiguration?: CredentialRevocationConfig
-  revocationStatusList?: RevocationStatusList
+  revocationStatusList?: RevocationStatusList | JsonObject
 }
 
 export type ProcessCredentialOptions = {
-  credentialRequestMetadata: CredentialRequestMetadata
-  masterSecret: MasterSecret
-  credentialDefinition: CredentialDefinition
-  revocationRegistryDefinition?: RevocationRegistryDefinition
+  credentialRequestMetadata: CredentialRequestMetadata | JsonObject
+  masterSecret: MasterSecret | JsonObject
+  credentialDefinition: CredentialDefinition | JsonObject
+  revocationRegistryDefinition?: RevocationRegistryDefinition | JsonObject
```

Comment:
The usage of `JsonObject` means we don't have any typing requirements for the object. Should it be `CredentialDefinition` or `CredentialDefinitionJson` where `CredentialDefinitionJson` specifies the structure the json needs to have?

Fine to not do that for now, but it feels like we're leaving out the most useful thing a wrapper can provide, which is a clear interface

Reply:
I agree. It would be more or less the same model we have in `@aries-framework/anoncreds`, right? IIRC there was still some work to do in the typing for some objects, so probably will take a bit more time to get it 100% accurate.